### ISSUE TITLE
GPT2ForSequenceClassification logits unmatched size

### ISSUE THREAD

**Issue by OshriAvnery** [huggingface/transformers#22324]
### System Info

huggingface-hub-0.13.3 tokenizers-0.13.2 transformers-4.27.2
python3.9

Hi :)

Using GPT2ForSequenceClassification, I have num_labels > 1, but get logits with shape (batch_size,1) instead of (batch_size, config.num_labels) as written in the docs.

I verified that those values are correct:
config.num_classes
model.num_labels
and model.score Linear with the correct out_features.

I also debugged and the shape of _logits_ inside the forward method is correct, so seems like the problem is with the line:
pooled_logits = logits[torch.arange(batch_size, device=logits.device), sequence_lengths]

maybe I didn't understand what it's supposed to do, but the result is getting logits in another shape than needed and than expected.

Will be glad for help, correction or clarification :)
Thank you!

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction
```python
config = GPT2Config(vocab_size = num_classes,
                    n_embd = input_size,
                    n_layer = 12,
                    n_head = 8,
                    num_labels = num_classes
                    )

model = GPT2ForSequenceClassification(config).to(device)
model.config.pad_token_id = model.config.eos_token_id

outputs = model(inputs_embeds=inputs).logits
```

### Expected behavior

outputs.shape == (batch_size, config.num_labels)
